src/clarity-angular/data/datagrid/datagrid.spec.ts

```diff
@@ -248,6 +248,28 @@ export default function(): void {
             });
         });
 
+        describe("Single selection", function() {
+            it("allows you to select and deselect a single row", function() {
+                const context = this.create(Datagrid, SingleSelectionTest);
+                expect(context.selected).toBeUndefined();
+                const radio = context.clarityElement.querySelectorAll(".datagrid-select input[type='radio']")[0];
+                const row = context.clarityElement.querySelectorAll(".datagrid-row")[1];
+                const foot = context.clarityElement.querySelector(".datagrid-foot-description");
+                expect(row.classList.contains("datagrid-selected")).toBeFalsy();
+                // Select a row
+                context.testComponent.selected = {value: 1};
+                radio.click();
+                context.detectChanges();
+                expect(row.classList.contains("datagrid-selected")).toBeTruthy();
```

Comment:
This unit test is really not "unit". The fact that you're grabbing shadow DOM elements, clicking on stuff, checking the result by looking up some class in the shadow DOM again, ... means your test is extremely fragile and is more akin to an end-to-end test.

You should use the two-way binding, check that the Selection provider of the Datagrid has been updated, and trust the unit test that check the click listeners, the view bindings and the dynamic CSS classes to do their job.

So your unit test would look like:
```typescript
it("allows to unselect in single selection mode", function() {
    const selection: Selection = context.getClarityProvider(Selection);
    selection.currentSingle = 2;
    context.detectChanges();
    context.testComponent.selected = null;
    context.detectChanges();
    expect(selection.currentSingle).toBeNull();
});
```

Reply:
I wrote it this way to track the bug that was found, where it would not fully unselect onclick, because the styling remained like it was active but the model could be set to `null`. The only way to test this is to inspect the DOM for classes to be applied or not. Your version is fine to test the bindings, but doesn't verify the bug was fixed.